### PR TITLE
Added Guild Features on Invite

### PR DESCRIPTION
**Pull request by Artuto**
[contributing]: https://github.com/DV8FromTheWorld/JDA/wiki/5%29-Contributing

## Pull Request Etiquette

There are several guidelines you should follow in order for your
  Pull Request to be merged.

- [x] I have checked the PRs for upcoming features/bug fixes.
- [x] I have read the [contributing guidelines][contributing].

> It is sometimes better to include more changes in a single commit. 
  If you find yourself having an overwhelming amount of commits, you
  can **rebase** your branch.

## Description

This PR adds a method to get the features of a guild on an invite.

### CODE REVIEW

**Comment by natanbc** [comment]
Why not create the set in EntityBuilder instead of creating a new one every time it's accessed?

**Comment by Artuto** [comment]
Didn't thought of that, done.

**Comment by kantenkugel** [comment]
Might be a good idea to wrap into a UnmodifiableSet and to make the json attribute optional (discord always gets us bad when deciding to just not send attributes sometimes)

**Comment by Artuto** [comment]
Discord docs says its not optional, and should appear on every guild object, but yea, discord is.. discord

**Comment by MinnDevelopment** [comment]
@Artuto the docs say in no way it should be in every guild object. Quite the opposite as their example does not include it. https://discordapp.com/developers/docs/resources/invite#invite-object-example-invite-object
The docs explicitly mention **partial** guild object

**Comment by Artuto** [comment]
Oh. I'll check if its present in the guild object then.

**Comment by kantenkugel** [comment]
And even if it would be in the docs... we had the case of a non-optional key being missing in the json payloads way to often, which broke JDA

**Comment by Artuto** [comment]
Should be ok now.

File: src/main/java/net/dv8tion/jda/core/entities/EntityBuilder.java [hunk 1]
```diff
@@ -1301,7 +1301,16 @@ public Invite createInvite(JSONObject object)
         final VerificationLevel guildVerificationLevel = VerificationLevel.fromKey(Helpers.optInt(guildObject, "verification_level", -1));
         final int presenceCount = Helpers.optInt(object, "approximate_presence_count", -1);
         final int memberCount = Helpers.optInt(object, "approximate_member_count", -1);
-        final Set<String> guildFeatures = StreamSupport.stream(guildObject.getJSONArray("features").spliterator(), false).map(String::valueOf).collect(Collectors.toSet());
+        final Set<String> guildFeatures;
+
+        if(guildObject.has("features"))
```
Comment (MinnDevelopment) [inline src/main/java/net/dv8tion/jda/core/entities/EntityBuilder.java]:
```java
if (guildObject.isNull("features"))
    guildFeatures = Collections.emptySet();
else
    guildFeatures = StreamSupport.stream(guildObject.getJSONArray("features").spliterator(), false).map(String::valueOf).collect(Collectors.toSet());
```

File: src/main/java/net/dv8tion/jda/core/entities/impl/InviteImpl.java [hunk 2]
```diff
@@ -303,7 +304,7 @@ public GuildImpl(final long id, final String iconId, final String name, final St
             this.verificationLevel = verificationLevel;
             this.presenceCount = presenceCount;
             this.memberCount = memberCount;
-            this.features = features;
+            this.features = Collections.unmodifiableSet(features);
```
Comment (MinnDevelopment) [inline src/main/java/net/dv8tion/jda/core/entities/impl/InviteImpl.java]: You are possibly wrapping the `Collections.emptySet()` in an unmodifiable decorator for no reason. Instead you should do this when you actually create a set.

**Comment by kantenkugel** [comment]
`isNull()` checks for non-existing as well, right? If so, this looks good to me.

**Comment by Sanduhr32** [comment]
Yes @kantenkugel. It checks whether an object is null and non-existing due `Map#get(key)` of a not existing entry returns `null`.
I think you even explained me the difference between a normal null check and the null check of `isNull()` and mentioned its checks.

**Pull request merged**